src/mdio/converters/segy_to_mdio_v1.py

```diff
@@ -0,0 +1,241 @@
+            err = f"Coordinate '{coord_name}' not found in SEG-Y dimensions."
+            raise ValueError(err)
+        coordinates.append(coord)
+
+    return dimensions, coordinates, index_headers
+
+def _populate_dims_coords_and_write_to_zarr(
+    dataset: xr_Dataset,
+    dimensions: list[Dimension],
+    coordinates: list[Dimension],
+    output: StorageLocation) -> xr_Dataset:
+    """Populate dimensions and coordinates in the xarray dataset and write to Zarr.
+
+    This will write the xr Dataset with coords and dimensions, but empty traces and headers.
+    """
+
+    # Should we do xarray.DataArray.reset_coords first?
+
+    vars_to_drop = list()
+
+    for c in dimensions:
+        dataset.coords[c.name] = c.coords
+        vars_to_drop.append(c.name)
+
+    for c in coordinates:
+        dataset.coords[c.name] = c.coords
+        vars_to_drop.append(c.name)
+
+    dataset.to_zarr(store=output.uri,
+                    mode="w",
+                    write_empty_chunks=False,
+                    zarr_format=2,
+                    compute=True)
+    
+    # Now we can drop them to simplify chunked write of the data variable
+    return dataset.drop_vars(vars_to_drop)
+
+def _create_grid_map_trace_mask(grid: Grid) -> tuple[zarr.Array, zarr.Array]:
+    """Create a trace mask and grid map arrays for the grid."""
+    # Tuple of arrays of integer indexes along each dimension for every live traces
+    live_indices = grid.header_index_arrays
+
+    # We set dead traces to uint32 max. Should be far away from actual trace counts.
+    grid_map = zarr.full(grid.shape[:-1], dtype="uint32", fill_value=UINT32_MAX)
+    grid_map.vindex[live_indices] = np.arange(live_indices[0].size)
+
+    trace_mask = zarr.zeros(grid.shape[:-1], dtype="bool")
+    trace_mask.vindex[live_indices] = 1
+
+    return grid_map, trace_mask
+
+def _populate_trace_mask_and_write_to_zarr(segy_file: SegyFile, 
+                                              dimensions: list[Dimension],
+                                              index_headers: SegyHeaderArray, 
+                                              xr_sd: xr_Dataset, 
+                                              output: StorageLocation) -> tuple[zarr.Array, xr_Dataset]:
+    """Populate and write the trace mask to Zarr, return the grid map as Zarr array.
+
+    Returns:
+      the grid map Zarr array.
+    """
+
+    # Create a grid and build live trace index
+    grid = Grid(dims=dimensions)
+    grid_density_qc(grid, segy_file.num_traces)
+    grid.build_map(index_headers)
+
+    # Create grid map and trace mask arrays
+    grid_map, trace_mask = _create_grid_map_trace_mask(grid=grid)
+    xr_sd.trace_mask.data[:] = trace_mask
+
+    # Populate the "trace_mask" variable in the xarray dataset and write it to Zarr
+    # Get subset of the dataset containing only "trace_mask"
+    #TODO: Should we write the trace mask inside of blocked_io.to_zarr?
+    #TODO: Should we clear the coords from ds_to_write?
```

Comment:
i think both are ok the way it is now in your code. no need to write these in blocked_io or clear coords; they're usually small.